addons/sourcemod/scripting/ConnectAnnounce.sp

```diff
@@ -515,35 +612,34 @@ stock Action SQLSelect_Join(Handle timer, any data)
 	pack.ReadCell();
 	pack.ReadString(sClientSteamID, sizeof(sClientSteamID));
 
-	Format(sQuery, sizeof(sQuery), "SELECT `message` FROM `join` WHERE `steamid` = '%s';", sClientSteamID);
-	if (g_hDatabase != null)
-		SQL_TQuery(g_hDatabase, OnSQLSelect_Join, sQuery, data, DBPrio_Low);
-	else
-		SetFailState("Database error while checking \"message\" from \"join\" table.");
-
-	return Plugin_Stop;
+	if (DB_Connect())
+	{
+		Format(sQuery, sizeof(sQuery), "SELECT `message` FROM `join` WHERE `steamid` = '%s';", sClientSteamID);
+		g_hDatabase.Query(OnSQLSelect_Join, sQuery, data);
+	}
 }
 
-public void OnSQLSelect_Join(Handle hParent, Handle hChild, const char[] err, any data)
+stock void OnSQLSelect_Join(Database db, DBResultSet results, const char[] error, any data)
 {
 	DataPack pack = view_as<DataPack>(data);
 	pack.Reset();
 
 	int client = pack.ReadCell();
+	delete pack;
 
-	if (hChild == null)
+	if (DB_Conn_Lost(results) || error[0] != '\0')
 	{
-		LogError("An error occurred while querying the database for the join message. (%s)", err);
+		LogError("%T (%s)", "Failed to query database", LANG_SERVER, error);
 		return;
 	}
-	else if (SQL_FetchRow(hChild))
+
+	if (results.FetchRow())
 	{
-		SQL_FetchString(hChild, 0, g_sClientJoinMessage[client], sizeof(g_sClientJoinMessage[]));
+		results.FetchString(0, g_sClientJoinMessage[client], sizeof(g_sClientJoinMessage[]));
+		delete results;
```

Comment:
delete after if statement because select might return 0 rows and handle will still exist